### PR TITLE
Remove twisted.python.win32.getProgramsMenuPath and twisted.python.win32.getProgramFilesPath

### PR DESCRIPTION
**Pull request by rodrigc**
https://twistedmatrix.com/trac/ticket/9312

twisted.python.win32.getProgramsMenuPath and twisted.python.win32.getProgramFilesPath
were deprecated in Twisted 15.3.0:
https://github.com/twisted/twisted/blob/trunk/NEWS.rst#deprecations-and-removals-21

and can now be removed, as specified in the Twisted Compatibility policy: https://twistedmatrix.com/documents/current/core/development/policy/compatibility-policy.html#incompatible-changes

### CODE REVIEW

**Comment by rodrigc** [comment]
@opalmer please review

**Review by opalmer — APPROVED** [review]
Looks good. I grepped around in the source code and don't see these used anywhere. I seem to recall asking about these functions a while back on the mailing list and the feedback was basically they're not used anywhere to the best of anyone's knowledge (I'd have to go looking for the thread to dig it up...)

**Review by opalmer — APPROVED** [review]

File: src/twisted/newsfragments/9312.removal [hunk 1]
```diff
@@ -0,0 +1 @@
+twisted.python.win32.getProgramsMenuPath and twisted.python.win32.getProgramFilesPath were deprectaed in Twisted 15.3.0 and have now been removed.
```
Comment (tdsmith) [inline src/twisted/newsfragments/9312.removal]: nit: s/deprectaed/deprecated/

Comment (opalmer) [reply src/twisted/newsfragments/9312.removal]: +1 agreed, I missed this.

**Review by opalmer — CHANGES_REQUESTED** [review]

**Review by opalmer — APPROVED** [review]

**Pull request merged**